### PR TITLE
Unable to rename category fields in the new content type portlet

### PR DESCRIPTION
**Pull request by alfredo-dotcms**

### CODE REVIEW

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/categories-property/categories-property.component.ts [hunk 1]
```diff
@@ -54,6 +58,15 @@ export class CategoriesPropertyComponent extends BaseComponent implements OnInit
         this.getCategoriesList(event.filter, event.first);
     }
 
+    /**
+     * Propagate the inode value to the parent form
+     * @param {any} categoryOptionSelected
+     * @memberof CategoriesPropertyComponent
+     */
+    categoryChanged(categoryOptionSelected: any): void {
```
Comment (fmontes) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/categories-property/categories-property.component.ts]: Any?

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/categories-property/categories-property.component.ts [hunk 2]
```diff
@@ -34,6 +36,8 @@ export class CategoriesPropertyComponent extends BaseComponent implements OnInit
             ? this.dotMessageService.get('contenttypes.field.properties.category.label')
             : this.property.value;
             this.paginationService.url = 'v1/categories';
+        this.groupLocalCopy = _.cloneDeep(this.group);
```
Comment (fmontes) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/categories-property/categories-property.component.ts]: Why? what's the problem here?

Comment (freddyucv) [reply src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/categories-property/categories-property.component.ts]: why do we need this copy?

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/categories-property/categories-property.component.html [hunk 3]
```diff
@@ -1,7 +1,8 @@
-<div [formGroup]="group">
+<div [formGroup]="groupLocalCopy">
     <label class="form__label">{{ i18nMessages['categories'] }}</label>
     <dot-searchable-dropdown #searchableDropdown 
         [data]="categoriesCurrentPage"
+        (change)="categoryChanged($event)"
```
Comment (freddyucv) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/categories-property/categories-property.component.html]: why the 'formControlName' doesn't work?

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/categories-property/categories-property.component.ts (hunk 2 above)
Comment (alfredo-dotcms) [reply src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/categories-property/categories-property.component.ts]: at this moment `categories` comes with a categories object (categoryName, description, inode, key, keywords, sortOrder) and to save it correctly we need on that property only the `inode`.

But we also need to pass that same categories object to the `dot-searchable-dropdown`, so I made a localcopy, to pass it to the dropdown and then update the parent categories attribute with the inode, so once it gets saved the endpoint doesn't break

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts [hunk 4]
```diff
@@ -97,6 +97,21 @@ export class ContentTypeFieldsPropertiesFormComponent extends BaseComponent impl
         });
     }
 
+    private formatFormData(): any {
+        const formCopy = Object.assign({}, this.form.value);
```
Comment (fmontes) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts]: If this object `this.form.value` have a nested object or collection this approach will NOT clone that, the reference to the nested object doesn't change, I think this is the case here, right?

**Pull request merged**